rclcpp_examples/CMakeLists.txt

```diff
@@ -59,52 +59,88 @@ if(AMENT_ENABLE_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 
-  # Add each test case.  Multi-executable tests can be specified in
-  # semicolon-separated strings, like  exe1;exe2.
-  set(example_tests
-    list_parameters_async
-    list_parameters
-    parameter_events_async
-    parameter_events
-    set_and_get_parameters_async
-    set_and_get_parameters
-    "talker:listener"
-    "add_two_ints_server:add_two_ints_client"
-    "add_two_ints_server:add_two_ints_client_async")
-
-  macro(tests)
-    foreach(example_test ${example_tests})
-      string(REPLACE ":" ";" example_executables "${example_test}")
+  # Test that the generated executables are linked against the specified rmw implementation
+  macro(test_linking)
+    set(executable_list
+      list_parameters_async
+      list_parameters
+      parameter_events_async
+      parameter_events
+      set_and_get_parameters_async
+      set_and_get_parameters
+      talker
+      listener
+      add_two_ints_server
+      add_two_ints_client
+      add_two_ints_client_async)
+
```

Comment:
is there anything I can call to access a list of all executables instead of writing them out myself?
I wanted to append names to a list when `custom_executable` is called, but I'm having issues with scope